nixos/modules/services/networking/netbird.nix

```diff
@@ -1,171 +1,383 @@
+                '';
+              };
+
+              name = mkOption {
+                type = str;
+                default = "netbird${config.suffix}";
+                defaultText = literalExpression ''netbird''${config.suffix}'';
+                description = ''
+                  Primary name for given client, used in naming of directories, services and users.
+                '';
+              };
+
+              interface = mkOption {
+                type = str;
+                default = "netbird${config.suffix}";
+                defaultText = literalExpression ''netbird''${config.suffix}'';
+                description = ''
+                  Name of the network interface managed by this client.
                 '';
               };
 
               environment = mkOption {
                 type = attrsOf str;
                 defaultText = literalExpression ''
                   {
-                    NB_CONFIG = "/var/lib/''${stateDir}/config.json";
-                    NB_LOG_FILE = "console";
-                    NB_WIREGUARD_PORT = builtins.toString port;
-                    NB_INTERFACE_NAME = name;
-                    NB_DAMEON_ADDR = "/var/run/''${stateDir}"
+                    NB_CONFIG = "/var/lib/''${config.name}/config.json";
+                    NB_DAEMON_ADDR = "unix:///var/run/''${config.name}/sock";
+                    NB_INTERFACE_NAME = config.interface;
+                    NB_LOG_FILE = mkOptionDefault "console";
+                    NB_LOG_LEVEL = config.logLevel;
+                    NB_WIREGUARD_PORT = toString config.port;
                   }
                 '';
                 description = ''
                   Environment for the netbird service, used to pass configuration options.
                 '';
               };
 
-              stateDir = mkOption {
-                type = str;
-                default = "netbird-${name}";
-                description = ''
-                  Directory storing the netbird configuration.
-                '';
+              autoStart = mkOption {
+                type = bool;
+                default = true;
+                description = "start service with the system";
+              };
+
+              hardened = mkOption {
+                type = bool;
+                default = true;
+                description = "harden the service by running as ephemeral user with minimal set of permissions";
+              };
+
+              logLevel = mkOption {
+                type = enum [
+                  # logrus loglevels
+                  "panic"
+                  "fatal"
+                  "error"
+                  "warn"
+                  "warning"
+                  "info"
+                  "debug"
+                  "trace"
+                ];
+                default = "info";
+                description = "log level of this client";
+              };
+
+              firewall.tcp.ports = mkOption {
```

Comment:
Why is the `firewall` option needed? The description is not informative

Reply:
The idea was to allow opening up firewall for services without having to figure out the interface naming conventions.

Having `config.interface` available makes it "good enough" to use with standard firewall module, so I could remove it.